data_steward/analytics/cdr_ops/rdr_export_qc.py

```diff
@@ -0,0 +1,307 @@
+#
+# will set variables like these which are used to construct SQL queries:
+#
+#     project_id = "my_project"
+#     old_rdr = "rdr1"
+#     new_rdr = "rdr2"
+#     cope_version_map = "dataset1.map_table"
+
+# +
+import urllib
+import pandas as pd
+
+NOTEBOOK_URL = None
+
+# Parameters required by this notebook
+REQUIRED_PARAMS = [
+    # identifies the google cloud project
+    'project_id',
+    # dataset_id of a prior RDR export for comparison
+    'old_rdr',
+    # dataset_id of the RDR export being evaluated
+    'new_rdr',
+    # dataset_id.table_id mapping questionnaire IDs to survey versions
+    'cope_version_map'
+]
+pd.options.display.max_rows = 120
+
+
+# -
+
+# ## Parse URL parameters
+# Retrieve parameters from the query string of the notebook URL. This allows sensitive
+# analytical parameters (e.g. project_id, dataset_id) to be specified ad hoc without
+# storing them in the notebook, which must be version controlled.
+#
+# **This approach to parameterizing notebooks has some major limitations and may be replaced in
+# the future.**
+# 1. The user must remember to run the following cells every time the query string is changed
+# 2. It may not be possible to specify the parameters this way when executing the notebook
+# programmatically (i.e. for automation)
+
+# + language="javascript"
+# // conditional needed to prevent warning
+# if (IPython.notebook.kernel) {
+#     // set the Python variable NOTEBOOK_URL
+#     IPython.notebook.kernel.execute("NOTEBOOK_URL = '" + window.location + "'");
+# }
+
+# +
+def get_url_params(url):
+    """
+    Extract params from the query string of a URL
+    
+    :returns: dict where the keys are parameter names
+    """
+    split_url = urllib.parse.urlsplit(url)
+    params = urllib.parse.parse_qs(split_url.query)
+    for key, value in params.items():
+        if len(value) == 1:
+            params[key] = value[0]
+    return params
+
+
+def validate_params(params, required_params):
+    """
+    Raise an error if a required parameter is missing from the params dict
+    """
+    for required_param in required_params:
+        if required_param not in params:
+            raise ValueError(f'Missing query parameter `{required_param}`')
+    print(f'All required parameters are specified.\n{params}')
+
+
+# -
+
+params = get_url_params(NOTEBOOK_URL)
+validate_params(params, REQUIRED_PARAMS)
+
+project_id = params['project_id']
+old_rdr = params['old_rdr']
+new_rdr = params['new_rdr']
+cope_version_map = params['cope_version_map']
+
+# # Missing tables
+
+query = f'''
+SELECT 
+ COALESCE(curr.table_id, prev.table_id) AS table_id, 
+ curr.row_count AS {new_rdr}, 
+ prev.row_count AS {old_rdr}, 
+ (curr.row_count - prev.row_count) row_diff 
+FROM {new_rdr}.__TABLES__ curr 
+FULL OUTER JOIN {old_rdr}.__TABLES__ prev
+  USING (table_id)
+WHERE curr.table_id IS NULL OR prev.table_id IS NULL
+'''
+pd.read_gbq(query, project_id=project_id, dialect='standard')
+
```

Comment:
Adding comments to tell me what I should expect as the result of running these queries would be helpful.  Right now, I wouldn't necessarily know how to interpret the results of any of these queries.